From 7e3db65608d811d792960c9320a8775fc9e03b0e Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 11 Jan 2025 19:20:43 +0100
Subject: [PATCH] Fix tests failing due to missing partial-pdm-dockerize.json
 schema

Tests running on the examples in tests/examples/store/ and
tests/invalid-examples/store/ currently fail with the following error
from fastjsonschema:

    KeyError: 'https://json.schemastore.org/partial-pdm-dockerize.json'

I'm not sure why fastjsonschema cannot resolve the remote schema
partial-pdm-dockerize.json. The reference chain seems to be:

pyproject.json -> partial-pdm.json -> partial-pdm-dockerize.json

Work-around the above by adding partial-pdm-dockerize.json to the
relevant test_config.json files.
---
 tests/examples/store/test_config.json         | 5 ++++-
 tests/invalid-examples/store/test_config.json | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/tests/examples/store/test_config.json b/tests/examples/store/test_config.json
index 116d760..13fd0b6 100644
--- a/tests/examples/store/test_config.json
+++ b/tests/examples/store/test_config.json
@@ -1,3 +1,6 @@
 {
-    "store": "https://json.schemastore.org/pyproject.json"
+    "store": "https://json.schemastore.org/pyproject.json",
+    "tools": {
+        "pdm-dockerize": "https://json.schemastore.org/partial-pdm-dockerize.json"
+    }
 }
diff --git a/tests/invalid-examples/store/test_config.json b/tests/invalid-examples/store/test_config.json
index 116d760..13fd0b6 100644
--- a/tests/invalid-examples/store/test_config.json
+++ b/tests/invalid-examples/store/test_config.json
@@ -1,3 +1,6 @@
 {
-    "store": "https://json.schemastore.org/pyproject.json"
+    "store": "https://json.schemastore.org/pyproject.json",
+    "tools": {
+        "pdm-dockerize": "https://json.schemastore.org/partial-pdm-dockerize.json"
+    }
 }
